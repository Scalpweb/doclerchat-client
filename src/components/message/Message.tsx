import React from 'react';
import {t} from 'components/translationContextGate/TranslationContextGate';
import styles from './Message.module.scss';
import {parseUrls} from '../../utils/utils';

interface Props {
  me?: boolean;
  info?: boolean;
  author: string;
  message: string;
  postedAt: Date;
  hour12: boolean;
}

const Message = ({me, info, author, message, postedAt, hour12}: Props) => {
  const formatMessage = (msg: string) => {
    if (info) {
      msg = msg.replace(/\$t\[(.*)\]/gim, (_, c) => t(c));
    } else {
      return parseUrls(msg);
    }
    return msg;
  };

  return (
    <div className={me ? styles.me : info ? styles.info : undefined}>
      <div className={styles.meta}>
        {author},{' '}
        {postedAt.toLocaleTimeString(navigator.language, {
          hour: '2-digit',
          minute: '2-digit',
          hour12: hour12
        })}
      </div>
      <div className={styles.message}>
        {message.split('\n').map((line, i) => (
          <span key={`line-${i}`}>
            {formatMessage(line)}
            <br />
          </span>
        ))}
      </div>
    </div>
  );
};

export default Message;
