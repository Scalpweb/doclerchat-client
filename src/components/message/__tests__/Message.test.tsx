import React from 'react';
import {shallow} from 'enzyme';
import Message from '../Message';

const date = new Date(0);

describe('Message', () => {
  it('renders with clock 12', () => {
    const component = shallow(<Message author="foo" message="bar" postedAt={date} hour12={true} />);
    expect(component).toMatchSnapshot();
  });

  it('renders with clock 24', () => {
    const component = shallow(
      <Message author="foo" message="bar" postedAt={date} hour12={false} />
    );
    expect(component).toMatchSnapshot();
  });
});
