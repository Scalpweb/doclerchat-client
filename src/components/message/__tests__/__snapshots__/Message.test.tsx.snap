// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message renders with clock 12 1`] = `
<div>
  <div>
    foo
    ,
     
    01:00 AM
  </div>
  <div>
    <span
      key="line-0"
    >
      <span>
        <Emoji
          key="bar"
          text="bar"
        />
      </span>
      <br />
    </span>
  </div>
</div>
`;

exports[`Message renders with clock 24 1`] = `
<div>
  <div>
    foo
    ,
     
    01:00
  </div>
  <div>
    <span
      key="line-0"
    >
      <span>
        <Emoji
          key="bar"
          text="bar"
        />
      </span>
      <br />
    </span>
  </div>
</div>
`;
